execution/src/transaction.rs

```diff
@@ -219,6 +210,12 @@ impl NormalizedExtendedTxEnvelope {
         self.tx_hash()
     }
 
+    /// RLP encoding suitable for computing the transaction root.
+    pub fn trie_encode(&self, out: &mut Vec<u8>) {
+        let op_envelope: OpTxEnvelope = self.clone().into();
+        op_envelope.encode_2718(out);
+    }
```

Comment:
We could also implement the `Encodable2718` / `Decodable2718` trait instead here and anywhere relevant, and I'd argue that would be a tiny bit cleaner. That would also give some default implemented methods for free, not that we need them now though.